protocol/decoder/decoder_interface.go

```diff
@@ -0,0 +1,42 @@
+package decoder
+
+// DecoderI is the interface providing helpers for reading with Kafka's encoding rules.
+// Types implementing Decoder only need to worry about calling methods like GetString,
+// not about how a string is represented in Kafka.
+type DecoderI interface {
```

Comment:
What's an example of where we're using `DecoderI` / `EncoderI`? Everything seems to work fine based on `Encoder` and `Decoder` directly, no?